blog/content/post/2020-01-15-GRUB2_and_3mdeb_minisummit.md

```diff
@@ -0,0 +1,177 @@
+ending up with unbootable product, which of course leads to all other problems
+on vendor side.
+
+We all agreed that this feature is needed in GRUB2 and if 3mdeb will get
+correct sponsoring from vendors using GRUB2 such support will be developed and
+contributed. On the other hand some vendors assume that file corruption is so
+rare that they don't want to spend additional engineering hours on such
+support.
+
+Before such support is available there are still methods to mitigate impact of
+the problem for which reference you can find in presentation.
+
+# TPM support in GRUB2 for legacy boot mode
+
+Key problem here is lack of boot process integrity in non-UEFI systems. Since
+GRUB2 is in boot process chain also in legacy systems we wanted to continue
+discussion started at LPC 2019. General conclusion on Linux Plumbers Conference
+was that approach presented in "TCG PC Client Specific Implementation
+Specification for Conventional BIOS" is preferred. What in short means that
+BIOS should expose legacy interrupt INT 1Ah to legacy operating system for
+measurement purposes.
+
+There are couple problems with that:
+* this approach is complaint with TPM1.2, no support for TPM2.0
+* GRUB2 is neither consumer or producer of INT 1Ah
+* at this point GRUB2 supports TPM only through UEFI API
+* GRUB2 should preserve interrupt handlers
+
+Because whole problem seems to not be so important (it affects limited amount
+of hardware) we are thinking about simplest way of having correct results. It
+seems that solution that has most sense is leveraging [Rhode and Schwarz TrustedGRUB2 implementation](https://github.com/Rohde-Schwarz/TrustedGRUB2).
+
+Key assumption of Rhode and Schwarz project is that BIOS measures MBR and
+interrupts for measuring bootloader kernel (`diskboot.img`) is called from MBR.
+In that way measurement chain continues. Of course something has to install
+those interrupts, in this case it is SeaBIOS, which implemented previous
+mentioned TCG specification.
+
+Addressing lack of support for TPM2.0 through INT 1Ah seems to not be big
+issues, since we can hide hardware that we dealing with in SeaBIOS
+implementation. Bootloader just need to extend PCRs and do not use any
+sophisticated TPM fores and int that area functionally there is not difference
+between TPM1.2 and TPM2.0. This approach implies that our stack needs SeaBIOS,
+since we were not able to find anyone who can confirm existence of legacy BIOS
+or CSM with INT 1Ah support. SeaBIOS of course limit us to coreboot and QEMU
+based platforms.
+
+Based on previous assumption we have to admit that GRUB2 can be only consumer
+of INT 1Ah API. Simplest solution for that would be to port support from
+TrustedGRUB2.
+
+In previously described cases GRUB2 is second stage bootloader. If we would
+like to have GRUB2 as first stage bootloader GRUB2 should be producer of INT
+1Ah API. That implies targets like:
+* \*BSD booted from GRUB2 on top of Legacy BIOS/CSM without INT 1Ah
```

Comment:
*BSD refers to the family of BSD based systems. the `\` is required in order to avoid formatting of `*`

Reply:
@artur-rs please confirm it will work

Reply:
@pietrushnic I can confirm that it will work, but I had to add missing empty line before the list for correct rendering (this info can be found in the post template)